### PR TITLE
feat: add Visual Editor/Code Editor Switch component

### LINKED ISSUE #673: Visual Editor / Code Editor Switch component
Implement the Switch component as seen in the [design](https://www.figma.com/file/bPMB3lkMTOOMuKk0oGLuNm/Studio?type=design&node-id=0%3A1&t=5t1QnTliI6risXyA-1).

Captura de pantalla 2023-06-16 a las 8 40 52

**Note:** Do not try to make it a generic Switch component. This one is a very special one with two icons so I would not call it `Switch` but, instead, `EditorSwitch` or something like that to avoid naming collisions with a future Switch component.

Related to #636

### LINKED ISSUE #692: Integrate eslint for style consistency
#### Reason/Context
Proposing the integration of ESLint into our project using the [existing configuration](https://github.com/asyncapi/studio/blob/master/apps/studio/.eslintrc) from the "studio" app.

### PR DESCRIPTION
**Pull request by KhudaDad414**
**Description**
* Add switch component.
* Add eslint script and config.
* Add `clean` script to remove all of the build and `node_modules` folders

**Related Issues**
Resolves #673
Resolves #692

### CODE REVIEW

File: packages/ui/components/EditorSwitch.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import React, { useState, useCallback } from 'react'
```
Comment (fmvilas) [inline packages/ui/components/EditorSwitch.tsx:1]: The component is not usable with the keyboard. My recommendation is that you wrap it with https://www.radix-ui.com/docs/primitives/components/switch.

**Review by fmvilas — CHANGES_REQUESTED** [review]
Left a suggestion. Other than that, looks good 👍

File: .eslintignore [hunk 2]
```diff
@@ -2,4 +2,5 @@ node_modules
 public
 docs
 lib
-build
\ No newline at end of file
+build
```
Comment (fmvilas) [inline .eslintignore:5]: Why did you move this file and `.eslintrc.json` to the root? 🤔

File: package.json [hunk 3]
```diff
@@ -5,9 +5,11 @@
     "url": "git://github.com/asyncapi/studio.git"
   },
   "scripts": {
+    "lint": "eslint",
```
Comment (fmvilas) [inline package.json:8]: Why do you need eslint here?

**Review by fmvilas — CHANGES_REQUESTED** [review]

File: .eslintignore (hunk 2 above)
Comment (KhudaDad414) [reply .eslintignore:5]: because `.eslintrc.json` wasn't picked up by eslint and we couldn't enforce code and formatting consistency across the packages.

File: package.json (hunk 3 above)
Comment (KhudaDad414) [reply package.json]: so that testing workflow can run the linter and ensure code consistency.

File: apps/design-system/src/components/OperationIcon.stories.tsx [hunk 4]
```diff
@@ -1,14 +1,15 @@
 import { OperationIcon } from '@asyncapi/studio-ui';
 
-export default {
-    component: OperationIcon,
-    parameters: {
-        backgrounds: {
-          default: 'dark'
-        }
-    }
+const meta = {
```
Comment (fmvilas) [inline apps/design-system/src/components/OperationIcon.stories.tsx:3]: What's the advantage of this change? 🤔 I honestly prefer the straight `export default` than having to define a variable and then export it.

File: .changeset/perfect-sloths-repeat.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+"@asyncapi/studio-ui": minor
+---
+
+Add EditorSwitch.
```
Comment (fmvilas) [inline .changeset/perfect-sloths-repeat.md:5]:
```suggestion
Add EditorSwitch component.
```

**Review by fmvilas — CHANGES_REQUESTED** [review]

File: apps/design-system/src/components/OperationIcon.stories.tsx (hunk 4 above)
Comment (KhudaDad414) [reply apps/design-system/src/components/OperationIcon.stories.tsx:3]: eslint recommends naming each export.
https://github.com/import-js/eslint-plugin-import/blob/v2.25.2/docs/rules/no-anonymous-default-export.md

I see we are not currently using that name in the  code but we might reuse some parts of the meta object in it's stories in other components. it would just add consistency between stories. but still, not a big deal 😆 I can disable this rule.

Comment (fmvilas) [reply apps/design-system/src/components/OperationIcon.stories.tsx:3]: 🤷‍♂️😄

**Review by fmvilas — DISMISSED** [review]

File: packages/ui/components/EditorSwitch.tsx [hunk 6]
```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+import * as Switch from '@radix-ui/react-switch'
```
Comment (Amzani) [inline packages/ui/components/EditorSwitch.tsx:2]: @KhudaDad414 @fmvilas Radix-ui need an ADR ? or this was already decided in the past ?

File: packages/ui/components/EditorSwitch.tsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+import { EyeIcon, CodeBracketIcon } from '../components/icons'
+import classNames from 'classnames'
+
+// Define styles as constants outside the component to avoid recreating these during each render
+const transitionStyle = 'transition duration-200 ease-in-out'
+const switchClass = 'relative bg-gray-800 flex h-[35px] w-[62px] cursor-pointer rounded'
```
Comment (Amzani) [inline packages/ui/components/EditorSwitch.tsx:8]: Variable naming consistency, we are using both *Style and *Class.

File: packages/ui/components/EditorSwitch.tsx (hunk 6 above)
Comment (KhudaDad414) [reply packages/ui/components/EditorSwitch.tsx:2]: I saw a few radix-ui components being used so I don't know 😆

Comment (fmvilas) [reply packages/ui/components/EditorSwitch.tsx:2]: I introduced it a few weeks ago so yeah, an ADR would be cool.

**Review by fmvilas — APPROVED** [review]

File: packages/ui/components/EditorSwitch.tsx (hunk 6 above)
Comment (Amzani) [reply packages/ui/components/EditorSwitch.tsx:2]: Created an issue for this: https://github.com/asyncapi/studio/issues/734
